nginx.conf.template

```diff
@@ -1,51 +1,76 @@
-# /etc/nginx/nginx.conf.template
-
+# /etc/nginx/nginx.conf
 user nginx;
-
-# Sets the number of worker processes based on CPU cores available
-worker_processes auto; 
+worker_processes auto; # Adjust based on the number of CPU cores
 pid /var/run/nginx.pid;
 
 events {
-    # Handles how many connections each worker process can handle simultaneously
-    worker_connections 1024; 
+    worker_connections 1024; # Maximum simultaneous connections per worker
 }
 
-# Handles requests for the Vue static files
 http {
-    # Loads the mime types file that defines mappings between file extensions and content types
-    include /etc/nginx/mime.types; 
-
-    # Default content type if unspecified
-    default_type application/octet-stream;
+    # Load standard MIME types
+    include /etc/nginx/mime.types;
+    default_type application/octet-stream; # Fallback MIME type for unknown files
 
-    # Enables the sendfile system call for faster file transfers
+    # Optimize file transfer
     sendfile on;
-
-    # Time in seconds to keep a connection open after completing a request
     keepalive_timeout 65;
 
+    # Define allowed CORS origins
+    map $http_origin $cors_origin {
+        default ""; # Default to disallow all origins
+        "http://localhost:5173" "http://localhost:5173";
+        "http://localhost:8080" "http://localhost:8080";
+        "http://127.0.0.1:5173" "http://127.0.0.1:5173";
+        "http://127.0.0.1:8080" "http://127.0.0.1:8080";
+        "https://__DEV_DOMAIN__" "https://__DEV_DOMAIN__";
+        "https://__PROD_DOMAIN__" "https://__PROD_DOMAIN__";
+    }
+
     server {
-        # Nginx listens on the specified port
         listen __PORT__; 
+        server_name __SERVER_NAME__; # Replace with the domain or hostname
```

Comment:
Put the magic word `frontend` instead of `__SERVER_NAME__` (that's what you name your web service in docker-compose in line 17), it will resolve itself to the frontend container IP address